template-manager-cmtemplate/src/test/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/impala/ImpalaVolumeConfigProviderTest.java

```diff
@@ -35,6 +37,21 @@ void testRoleConfigsWithAttachedVolumes100GB() {
         );
     }
 
+    @Test
+    void testRoleConfigsWithAttachedVolumes100GBAndTemporaryStoragePresent() {
+        HostgroupView worker = hostGroupWithVolumeTemplatesAndTemporaryStorage(2, getVolumeTemplates(100), TemporaryStorage.EPHEMERAL_VOLUMES, 3);
+        List<ApiClusterTemplateConfig> roleConfigs = subject.getRoleConfigs(ImpalaRoles.ROLE_IMPALAD, worker, preparatorWithHostGroups(worker));
+
+        assertEquals(
+                List.of(
+                        config("scratch_dirs", "/hadoopfs/ephfs1/impala/scratch,/hadoopfs/ephfs2/impala/scratch,/hadoopfs/ephfs3/impala/scratch"),
+                        config("datacache_enabled", "true"),
+                        config("datacache_capacity", "53687091200"),
```

Comment:
It looks like the value of 50 GB here for datacache_capacity is based on a minimum volume size of 100 GB / 2.  Once my comment above is addressed about calculating the size based on the min of either ephemeral or attached, this may need to be adjusted.  Also I can't tell how the test code would actually determine the size of ephemeral volumes since AFAICT we're not planning to explicitly specify the ephemeral volume sizes in the templates (so presumably CB will get the ephemeral sizes based on the instance type?).

Reply:
@drorke  Yes, CB will get the ephemeral sizes based on the instance type.
To calculate `minAttachedVolumeSize` both attached volumes and ephemeral disks size are considered.